### ISSUE TITLE
Developer documentation missing or hard to find

### ISSUE THREAD

**Issue by bradneuman** [OptiKey/OptiKey#773]
**Is your feature request related to a problem? Please describe.**
I'm a software engineer but not particularly familiar with .NET development. I read through all the readme docs I could find, including CONTRIBUTING and didn't find any instructions / quick start on developer documentation. What tools do I need? How do I open and build the project? For example, can I use VSCode or do I need visual studio / some other SDK?

**Describe the solution you'd like**
I'd like to see a "developer" section in the docs folder (or elsewhere) with clear links from the contributing page.

**Describe alternatives you've considered**
Google search didn't turn up much.

**Additional context**
I plan to add this documentation and solve this issue myself, just opening to get attention / see if I'm missing something. Hopefully this will be my first PR to the project :)

**Comment by bradneuman** [comment]
(I have figured out that I think I need Visual Studio, and was able to build and run the project, will add some docs once I feel like I know what I'm doing a tiny bit more)

**Comment by AdamRoden** [comment]
@bradneuman I might be able to answer some questions for you. I used to be an active contributor, but it has been a while.

**Comment by bradneuman** [comment]
Thank you @AdamRoden! Right now I've gotten the project building and am working on a PR with some documentation on the general project file structure -- I'll send it your way when I have a draft ready and you can tell me if I'm totally off base? Do you know if there's anything already written up about the general structure of the code, tech used, etc? I found the "design decisions" doc but it's pretty sparse

**Comment by AdamRoden** [comment]
@bradneuman I do not believe there is any documentation around structure of the code. As such, it can be challenging for someone new to learn how various pieces work together. Once you send something I will look it over and ensure you're on the right track.

**Comment by tqphan** [comment]
@bradneuman Your efforts will help me as well. I've always wanted to contribute to this project programmatically but lack the dedication and knowledge to figure out how different parts of the project work together.
